### PR TITLE
Delivery API nested property expansion and output limiting

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

TBD 😄

### CODE REVIEW

File: src/Umbraco.Cms.Api.Delivery/Controllers/Media/ByIdMediaApiController.cs [hunk 1]
```diff
@@ -9,23 +9,35 @@
...
     /// </summary>
     /// <param name="id">The unique identifier of the media item.</param>
     /// <returns>The media item or not found result.</returns>
     [HttpGet("item/{id:guid}")]
-    [MapToApiVersion("1.0")]
+    [MapToApiVersion("2.0")]
     [ProducesResponseType(typeof(IApiMediaWithCropsResponse), StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> ById(Guid id)
+    public async Task<IActionResult> ById20(Guid id)
```
Comment (elit0451) [inline src/Umbraco.Cms.Api.Delivery/Controllers/Media/ByIdMediaApiController.cs:37]:
```suggestion
    public async Task<IActionResult> ByIdV20(Guid id)
```

File: src/Umbraco.Cms.Api.Delivery/Filters/SwaggerDocumentationFilterBase.cs [hunk 2]
```diff
@@ -60,19 +104,52 @@ protected void AddQueryParameterDocumentation(OpenApiParameter parameter, Dictio
...
-
-    private string QueryParameterDescription(string description)
-        => $"{description}. Refer to [the documentation]({DocumentationLink}#query-parameters) for more details on this.";
+    private void AddFields(OpenApiOperation operation)
+        => operation.Parameters.Add(new OpenApiParameter
+        {
+            Name = "fields",
+            In = ParameterLocation.Query,
+            Required = false,
+            Description = QueryParameterDescription("Explicitly defines which properties that should be included in the response (by default all properties are included)"),
```
Comment (elit0451) [inline src/Umbraco.Cms.Api.Delivery/Filters/SwaggerDocumentationFilterBase.cs:140]:
```suggestion
            Description = QueryParameterDescription("Explicitly defines which properties should be included in the response (by default all properties are included)"),
```

File: src/Umbraco.Cms.Api.Delivery/Filters/SwaggerDocumentationFilterBase.cs [hunk 3]
```diff
@@ -60,19 +104,52 @@ protected void AddQueryParameterDocumentation(OpenApiParameter parameter, Dictio
...
+            Schema = new OpenApiSchema { Type = "string" },
+            Examples = new Dictionary<string, OpenApiExample>
+            {
+                { "Include all properties", new OpenApiExample { Value = new OpenApiString("properties[$all]") } },
+                {
+                    "Include only specific properties",
+                    new OpenApiExample { Value = new OpenApiString("properties[alias1,alias2]") }
+                },
```
Comment (elit0451) [inline src/Umbraco.Cms.Api.Delivery/Filters/SwaggerDocumentationFilterBase.cs:148]: Adding an example with a single property, to align with the `expand` examples
```suggestion
                {
                    "Include only specific property",
                    new OpenApiExample { Value = new OpenApiString("properties[alias1]") }
                },
                {
                    "Include only specific properties",
                    new OpenApiExample { Value = new OpenApiString("properties[alias1,alias2]") }
                },
```

**Review by elit0451 — CHANGES_REQUESTED** [review]
I've added a few minor suggestions for you to consider

**Comment by kjac** [comment]
Good stuff @elit0451 👍

**Pull request merged**